packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -1159,4 +1162,101 @@ public ResponseEntity<Void> cancelTask(@PathVariable("task-id") final UUID taskI
 		taskService.cancelTask(TaskType.GOLLM, taskId);
 		return ResponseEntity.ok().build();
 	}
+
+	@PostMapping("/document-question")
+	@Secured(Roles.USER)
+	@Operation(summary = "Dispatch a `GoLLM Document Question` task")
+	@ApiResponses(
+		value = {
+			@ApiResponse(
+				responseCode = "200",
+				description = "Dispatched successfully",
+				content = @Content(
+					mediaType = "application/json",
+					schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = TaskResponse.class)
+				)
+			),
+			@ApiResponse(
+				responseCode = "422",
+				description = "The request was interrupted while waiting for a response",
+				content = @Content
+			),
+			@ApiResponse(
+				responseCode = "503",
+				description = "The request was timed out while waiting for a response",
+				content = @Content
+			),
+			@ApiResponse(responseCode = "500", description = "There was an issue dispatching the request", content = @Content)
+		}
+	)
+	public ResponseEntity<TaskResponse> createDocumentQuestionTask(
+		@RequestParam(name = "mode", required = false, defaultValue = "SYNC") final TaskMode mode,
+		@RequestParam(name = "project-id", required = false) final UUID projectId,
+		@RequestParam(name = "document-id", required = true) final UUID documentId,
+		@RequestBody final String question
+	) {
+		// create the task
+		final TaskRequest req = new TaskRequest();
+		req.setType(TaskType.GOLLM);
+		req.setScript(DocumentQuestionHandler.NAME);
+		req.setUserId(currentUserService.get().getId());
+
+		final DocumentQuestionHandler.Input input = new DocumentQuestionHandler.Input();
+		input.setLlm(config.getLlm());
+		input.setQuestion(question);
+
+		// Grab the document
+		DocumentAsset document = documentAssetService
+			.getAsset(documentId)
+			.orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("document.not-found")));
+
+		// make sure there is a text in the document
+		if (document.getText() == null || document.getText().isBlank()) {
```

Comment:
You are setting input to `document.getExtractions()` but you are checking for `document.getText()`

Reply:
fixed